tests/tensorflow2/test_keras.py

```diff
@@ -238,3 +242,57 @@ def test_weights_collections(out_dir, tf_eager_mode):
     assert len(trial.tensor_names(collection=CollectionKeys.WEIGHTS)) == 2
     assert len(trial.tensor_names(collection=CollectionKeys.LOSSES)) == 1
     assert len(trial.tensor_names(collection=CollectionKeys.METRICS)) == 3
+
+
+@pytest.mark.slow
+def test_include_collections(out_dir, tf_eager_mode):
+    include_collections = [
+        CollectionKeys.WEIGHTS,
+        CollectionKeys.BIASES,
+        CollectionKeys.GRADIENTS,
+        CollectionKeys.LOSSES,
+        CollectionKeys.OUTPUTS,
+        CollectionKeys.METRICS,
+        CollectionKeys.OPTIMIZER_VARIABLES,
+    ]
+    save_config = SaveConfig(save_interval=3)
+    hook = smd.KerasHook(
+        out_dir,
+        save_config=save_config,
+        include_collections=include_collections,
+        reduction_config=ReductionConfig(norms=ALLOWED_NORMS, reductions=ALLOWED_REDUCTIONS),
+    )
+    helper_keras_fit(out_dir, hook=hook, steps=["train", "eval", "predict"], eager=tf_eager_mode)
+
+    trial = smd.create_trial(path=out_dir)
+    # can't save gradients in TF 2.x
+    if tf_eager_mode:
+        assert len(trial.tensor_names()) == 8
+    else:
+        assert len(trial.tensor_names()) == 18
+        assert len(trial.tensor_names(collection=CollectionKeys.GRADIENTS)) == 4
+        assert len(trial.tensor_names(collection=CollectionKeys.OPTIMIZER_VARIABLES)) == 5
+    assert len(trial.tensor_names(collection=CollectionKeys.BIASES)) == 2
+    assert len(trial.tensor_names(collection=CollectionKeys.WEIGHTS)) == 2
+    assert len(trial.tensor_names(collection=CollectionKeys.LOSSES)) == 1
+    assert len(trial.tensor_names(collection=CollectionKeys.METRICS)) == 3
+
+
+@pytest.mark.slow
+def test_hook_from_json(tf_eager_mode, monkeypatch):
+    out_dir = "/tmp/test"
+    shutil.rmtree(out_dir, ignore_errors=True)
```

Comment:
Might want to improve the cleanup logic here - use a TemporaryDirectory context manager? https://docs.python.org/3/library/tempfile.html . This test will leave files left over.

Reply:
done in the latest commit